src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -724,11 +720,11 @@ struct KeySetData : PersistentData<kPersistentBufferMax>
         fabric_index(fabric), keyset_id(id), policy(policy_id), keys_count(num_keys)
     {}
 
-    CHIP_ERROR UpdateKey(DefaultStorageKeyAllocator & key) override
+    CHIP_ERROR UpdateKey(StorageKey & key) override
     {
         VerifyOrReturnError(kUndefinedFabricIndex != fabric_index, CHIP_ERROR_INVALID_FABRIC_INDEX);
         VerifyOrReturnError(kInvalidKeysetId != keyset_id, CHIP_ERROR_INVALID_KEY_ID);
-        key.FabricKeyset(fabric_index, keyset_id);
+        key = DefaultStorageKeyAllocator::FabricKeyset(fabric_index, keyset_id);
```

Comment:
This now does a copy of the string instead of the in-place bit it used to do, right?  How does codesize generally look for this change?

Reply:
From what I could see, DefaultStorageKeyAllocator used to contain a buffer of data + pointer at all times, so there seemed to be little space saving overall. 

Looking at the bot reports, debug lines increased a lot, but code size not so much and it seems we even may have decreased some bss and text on some platforms. Unsure how much to trust it, but generally I believe this should not have large impact: code is almost the same, it does do a strcpy at times instead of a pointer assignment and it lowers the storage requirements by omiting the pointer (always use the embedded string since we have it).